File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/events/PlayerDriftStateEvent.java
package com.bitfire.uracer.game.events;


public final class PlayerDriftStateEvent extends
        Event<PlayerDriftStateEvent.Type, PlayerDriftStateEvent.Order> {
    public PlayerDriftStateEvent() {
        super(Type.class, Order.class);
    }

    public enum Type {
        onBeginDrift, onEndDrift
    }

    public enum Order {
        Default
    }

    public interface Listener extends Event.Listener<Type, Order> {
        @Override
        void handle(Object source, Type type, Order order);
    }
}

File: DriftGame/uracer-desktop/src/com/bitfire/uracer/game/events/PlayerDriftStateEvent.kt
package com.bitfire.uracer.game.events

class PlayerDriftStateEvent : Event<PlayerDriftStateEvent.Type, PlayerDriftStateEvent.Order>(Type::class.java, Order::class.java) {

    enum class Type {
        OnBeginDrift, OnEndDrift
    }

    enum class Order {
        Default
    }

    fun interface Listener : Event.Listener<Type, Order> {
        override fun handle(source: Any, type: Type, order: Order)
    }
}
